packages/app/src/cli/models/extensions/specifications/transform/app_config_webhook.ts

```diff
@@ -1,87 +1,46 @@
-  const webhooks = getPathValue<WebhooksConfig>(content, 'webhooks')
+  const webhooks = getPathValue(content, 'webhooks') as WebhooksConfig
   if (!webhooks) return content
 
-  // normalize webhook config with the top level config
   const webhookSubscriptions = []
-  const {topics, subscriptions, uri, api_version: apiVersion} = webhooks
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  const {api_version, subscriptions = []} = webhooks
 
-  if (uri && topics?.length) {
-    for (const topic of topics) {
-      webhookSubscriptions.push({
-        topic,
-        uri,
-      })
-    }
+  // eslint-disable-next-line no-warning-comments
+  // TODO: pass along compliance_topics once we're ready to store them in its own module
+  for (const {uri, topics, compliance_topics: _, ...data} of subscriptions) {
+    webhookSubscriptions.push(topics.map((topic) => ({uri, topic, ...data})))
   }
 
-  if (subscriptions?.length) {
-    for (const {path, uri: localUri, ...subscription} of subscriptions) {
-      // we can assume this is valid from earlier validation, and local URI will overwrite top level if there is any
-      const subscriptionConfig = {
-        uri: localUri || uri,
-        ...subscription,
-      }
-
-      if (path) {
-        subscriptionConfig.uri = `${subscriptionConfig.uri}${path}`
-      }
-
-      webhookSubscriptions.push(subscriptionConfig)
-    }
-  }
-  return webhookSubscriptions.length > 0
-    ? {api_version: apiVersion, subscriptions: webhookSubscriptions}
-    : {api_version: apiVersion}
+  return webhookSubscriptions.length > 0 ? {subscriptions: webhookSubscriptions.flat(), api_version} : {api_version}
 }
 
 export function transformToWebhookConfig(content: object) {
   let webhooks = {}
   const apiVersion = getPathValue(content, 'api_version') as string
   webhooks = {...(apiVersion ? {webhooks: {api_version: apiVersion}} : {})}
-  const serverWebhooks = getPathValue<WebhookSubscription[]>(content, 'subscriptions')
+  const serverWebhooks = getPathValue(content, 'subscriptions') as NormalizedWebhookSubscription[]
   if (!serverWebhooks) return webhooks
 
-  const frequencyMap: {[key: string]: number} = {}
-  serverWebhooks.forEach((item) => {
-    frequencyMap[item.uri!] = (frequencyMap[item.uri!] || 0) + 1
-  })
-  const maxCount = Math.max(...Object.values(frequencyMap))
-  const defaultUri = Object.keys(frequencyMap).find((key) => frequencyMap[key] === maxCount)
-
-  const topics: string[] = []
-  const subscriptions: WebhookSubscription[] = []
+  const webhooksSubscriptionsConfig: WebhooksConfig['subscriptions'] = []
 
-  for (const item of serverWebhooks) {
-    if (item.uri === defaultUri && !item.sub_topic && !item.include_fields && !item.metafield_namespaces) {
-      topics.push(item.topic)
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  for (const {uri, topic, sub_topic, ...optionalFields} of serverWebhooks) {
```

Comment:
> opted to spread (...) the rest of the subscription params in case we ever support more, so we won't need to update all the parsing logic and only the schema + module (only the module in the future I guess once CLI as a client is a thing)

Makes sense, it also gets simpler. But I've renamed `data` to `optionalFields`, which sounds more descriptive to me. Is it ok?